### ISSUE TITLE
IXWebSocket will timeout 100% of the time if the ping interval and ping timeout are the same amount of time.

### ISSUE THREAD

**Issue by maxweisel** [machinezone/IXWebSocket#165]
Running the following will cause a connection close after five seconds. I believe it's because the first ping is sent out after five seconds and the first timeout check also happens at five seconds. In practice, I imagine the ping should be sent immediately on connect so it has time to return before the first timeout check.

```
WebSocket webSocket;
webSocket.setPingInterval(5);
webSocket.setPingTimeout(5);
webSocket->setUrl("<URL>");
webSocket->start();
```

I think I can fix this one easily. I'm going to try to write up a PR real quick.

Max

**Comment by maxweisel** [comment]
Opened a PR here: https://github.com/machinezone/IXWebSocket/pull/166

**Comment by bsergean** [comment]
Thanks, looking at it now.

**Comment by bsergean** [comment]
Should be fixed in 8.3
